### ISSUE TITLE
Annotations on annotation properties are not working

### ISSUE THREAD

**Issue by Sandared** [eclipse/xtext-xtend#206]
```
@ObjectClassDefinition
annotation MyConfiguration{

    @AttributeDefinition
    String someProp = 'test'

}
```

Xtend complains about "The annotation @AttributeDefinition is disallowed for this location", but if I modify the generated Java file there is no error shown:

```
@ObjectClassDefinition
public @interface MyConfiguration {

    @AttributeDefinition
    public String someProp() default "test";

}
```

**Comment by cdietrich** [comment]
annotation looks like
```
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(value = RetentionPolicy.CLASS)
@Target(value = ElementType.METHOD)
public @interface AttributeDefinition {
}

```

xtend validator does

```
final EObject eContainer = getContainingAnnotationTarget(annotation);
		for (Entry<Class<?>, Collection<ElementType>> mapping : targetInfos.asMap().entrySet()) {
```

eContainer (=the annotated object) is a XtendField.

the map is populated as:

```
result.put(XtendField.class, ElementType.FIELD);
```

`ElementType.FIELD` != `ElementType.METHOD`

=> i guess we need some special treatment for xtendfields in annotations

**Comment by cdietrich** [comment]
fixed in xtext 2.12
